### PR TITLE
containerized provisioning environment

### PR DESCRIPTION
**Pull request by tourunen**
This commit adds pac-deployer, a container with all external software
dependencies included for running deployment. Scripts for local builds
as well as creating a build config in existing OpenShift are included
(the latter is a consolidated version of existing build config script
for oso-default-www).

The documentation in README.md is updated to include running
containerized deployment for heat_provisioning playbook.

Master copy of python dependency file requirements.txt is now held in
container-src/pac-deployer/ and a symlink exists for backwards
compatibility in playbooks/openshift/.

### CODE REVIEW

**Comment by rlaurika** [comment]
@tourunen I guess the container, once deployed, doesn't need to be updated all that often? Can just run on the existing one and make changes to the code that's mounted there?

File: playbooks/openshift/README.md [hunk 1]
```diff
@@ -212,10 +242,18 @@ For initialize_ramdisk.yml to work, you will need to populate the following vari
...
   * openshift_cloudprovider_openstack_tenant_id
   * openshift_cloudprovider_openstack_tenant_name
-  * penshift_cloudprovider_openstack_region
+  * openshift_cloudprovider_openstack_region
 
 Once you have all of this configured, running the actual installation is simple.
 
+Change the current working directory to playbooks/openshift:
+
+    $ cd ̃/git/pouta-ansible-cluster/playbooks/openshift
```
Comment (tourunen) [inline playbooks/openshift/README.md]: Fixed. In my editor it was two different characters, but tilde was indeed superscripted.

File: playbooks/openshift/README.md [hunk 2]
```diff
@@ -93,6 +94,34 @@ For automatic, self-provisioned app routes to work, you will need a wildcard DNS
...
+
+We also have a deployment container with all dependencies preinstalled. To build the container, 
+check out PAC git repo (see below) and run the build script located in `container-src/pac-deployer`:
+ 
+    cd ~/git/pouta-ansible-cluster/container-src/pac-deployer 
+    ./build.bash
+    
+To launch a shell in a temporary container for deployment, run    
+    
+    cd ~/git
```
Comment (tourunen) [inline playbooks/openshift/README.md]: Made a comment on using sudo if necessary for both build and run phases. Added a helper script.

File: playbooks/openshift/README.md [hunk 3]
```diff
@@ -93,6 +94,34 @@ For automatic, self-provisioned app routes to work, you will need a wildcard DNS
  
 In general, see https://docs.openshift.org/latest/install_config/install/prerequisites.html
 
+#### Alternative 2: containerized environment for Heat based deployment
+
+We also have a deployment container with all dependencies preinstalled. To build the container, 
+check out PAC git repo (see below) and run the build script located in `container-src/pac-deployer`:
+ 
+    cd ~/git/pouta-ansible-cluster/container-src/pac-deployer 
+    ./build.bash
```
Comment (tourunen) [inline playbooks/openshift/README.md]: See below.

File: container-src/pac-deployer/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,28 @@
+FROM centos:7
```
Comment (rlaurika) [inline container-src/pac-deployer/Dockerfile]: We could think about making this based on alpine at some point to make it a bit lighter.

**Pull request merged**